pkg/deploy/deploy.go

```diff
@@ -488,4 +388,132 @@ func OperatorExists(cli client.Client, operatorPrefix string) (bool, error) {
 	return false, nil
 }
 
+func getResource(ctx context.Context, cli client.Client, obj *unstructured.Unstructured) (*unstructured.Unstructured, error) {
+	found := obj.DeepCopy()
```

Comment:
Why it was done such a way at the first place? Get() does not need the object from manifests. But the logic was "if it is not deleted, not 'not found', but we could not fetch it, continue patching with the data we got from the manifests, not from the cluster".

Reply:
Does
```
func getResource(ctx context.Context, cli client.Client, obj *unstructured.Unstructured) (*unstructured.Unstructured, error) {
	found := &unstructured.Unstructured{}
	err := cli.Get(ctx, types.NamespacedName{Name: obj.GetName(), Namespace: obj.GetNamespace()}, found)
	if err != nil {
		return nil, err
	}
	return found, nil
}
```
makes it work as expected or I'm missing something?